galsim/wfirst/wfirst_backgrounds.py

```diff
@@ -0,0 +1,175 @@
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""
+@file wfirst_backgrounds.py
+
+Part of the WFIRST module.  This file includes any routines needed to define the background level,
+for which the main contribution is zodiacal light.
+"""
+
+import galsim
+import numpy as np
+import os
+
+def getSkyLevel(bandpass, position=None, e_lat=None, e_lon=None, exp_time=None):
+    """
+    Get the expected sky level for a WFIRST observation due to zodiacal light for this bandpass and
+    position.
+
+    This routine requires Bandpass objects that were loaded by galsim.wfirst.getBandpasses().  That
+    routine will have stored tables containing the sky background as a function of position on the
+    sky for that bandpass.  This routine then interpolates between the values in those tables to
+    arbitrary positions on the sky.
+
+    The numbers that are stored in the Bandpass object `bandpass` are background level in units of
+    e-/m^2/s/arcsec^2.  To get rid of the m^2, this routine multiplies by the total effective
+    collecting area in m^2, and to get rid of the arcsec^2 it multiplies by the pixel area in
+    arcsec^2.  This will give a result in e-/s/pix.  If the user has passed in an exposure time, then
+    the results are returned in e-/pix, otherwise the results are returned in e-/pix/s and the user
+    must multiply by the exposure time later on themselves.
+
+    The source of the tables that are being interpolated is Chris Hirata's publicly-available WFIRST
+    exposure time calculator (ETC):
+
+        http://www.tapir.caltech.edu/~chirata/web/software/space-etc/
+
+    It nominally returns photons/m^2/s/arcsec^2, but the input bandpasses used internally by the ETC
+    code include the quantum efficiency, to effectively convert to e-/m^2/s/arcsec^2.
+
+    Positions can be specified either with a keyword `position`, which must be a CelestialCoord
+    object, or with both `e_lat` and `e_lon` for the ecliptic coordinates (as Angle instances).
+    Without any of those values supplied, the routine will use a default position that looks
+    sensibly away from the sun.
+
+    @param bandpass     A Bandpass object.
+    @param position     A position, given as a CelestialCoord object.  If None, and `e_lat` and
```

Comment:
So far, most galsim functions that take a position call the argument `pos` rather than `position`.  Another possibility would be `world_pos` to follow the convention used in wcs.py for CelestialCoord positions to emphasize that it is not the position on the chip, but rather the world coordinate position.